Documentation/ContentObjects/Fluidtemplate/DataProcessing/FilesProcessor.rst

```diff
@@ -0,0 +1,242 @@
+   :default: "files"
+
+   The variable's name to be used in the Fluid template
+
+.. confval:: dataProcessing
+
+   :Required: false
+   :type: array of :ref:`dataProcessing`
+   :default: ""
+
+   Array of DataProcessors to be applied to all fetched records.
+
+
+Example 1: Render the images stored in field image
+==================================================
+
+Please see also :ref:`dataProcessing-about-examples`.
+
+
+TypoScript
+----------
+
+Using the :php:`FilesProcessor` the following scenario is possible::
+
+   tt_content {
+      examples_dataprocfiles =< lib.contentElement
+      examples_dataprocfiles {
+         templateName = DataProcFiles
+         dataProcessing.10 = TYPO3\CMS\Frontend\DataProcessing\FilesProcessor
+         dataProcessing.10 {
+            as = images
+            references.fieldName = image
+            references.table = tt_content
+            sorting = title
+            sorting.direction = descending
+         }
+      }
+   }
+
+
+The Fluid template
+------------------
+
+In the Fluid template then iterate over the files:
+
+.. code-block:: html
+
+   <html data-namespace-typo3-fluid="true" xmlns:f="http://typo3.org/ns/TYPO3/CMS/Fluid/ViewHelpers">
+      <h2>Data in variable images</h2>
+      <f:debug inline="true">{images}</f:debug>
+      <h2>Data in variable images</h2>
+
+      <div class="row">
+         <div class="row">
+            <f:for each="{images}" as="image">
+               <div class="col-12 col-md-3">
+                  <div class="card">
+                     <f:image image="{image}" class="card-img-top" height="250"/>
+                     <div class="card-body">
+                        <h5 class="card-title">{image.title}</h5>
+                        <div class="card-text">{image.description}</div>
+                     </div>
+                  </div>
+               </div>
+            </f:for>
+         </div>
+      </div>
+   </html>
+
+
+Output
+------
+
+The array :php:`images` contains the data of the files now.
+
+.. image:: Images/FilesProcessor.png
+   :class: with-shadow
+   :alt: files dump and output
+
+.. note::
+
+   For technical reasons FileReferences do not show all available data on
+   using debug. See :ref:`t3coreapi:fal-using-fal-frontend`.
+
+
+Example 2: use stdWrap property on references
+=============================================
+
+.. versionadded:: 10.3
+
+   :ref:`stdWrap` properties added to the :php:`FilesProcessor`.
+
+The following example implements a slide-functionality on rootline
+for file resources::
+
+   page.10.dataProcessing {
+      10 = TYPO3\CMS\Frontend\DataProcessing\FilesProcessor
+      10 {
+         references.data = levelmedia: -1, slide
+         as = myfiles
+      }
+   }
+
+The :php:`FilesProcessor` can slide up the rootline to collect
+images for FLUID templates. One usual feature is to use images
+attached to pages and use them down the page tree for
+header images in the frontend.
```

Comment:
```suggestion
images for FLUID templates. One usual feature is to take images
attached to pages and use them on the page tree as
header images in the frontend.
```
I tried to make sense of this sentence but I don't think I got it - can you fix?